core/src/main/scala/kafka/server/metadata/ZkMetadataCache.scala

```diff
@@ -376,6 +436,24 @@ class ZkMetadataCache(
   // This method returns the deleted TopicPartitions received from UpdateMetadataRequest
   def updateMetadata(correlationId: Int, updateMetadataRequest: UpdateMetadataRequest): Seq[TopicPartition] = {
     inWriteLock(partitionMetadataLock) {
+      if (updateMetadataRequest.updateType() == AbstractControlRequest.Type.FULL) {
+        if (updateMetadataRequest.version() < 8) {
```

Comment:
Seems a bit weird to check this. Doesn't the schema prevent this?

Reply:
Uh, yes. That's a good point (I was thinking about LISR which has had this field for a while). I'll remove this check